File: server/domain/repositories/account_repository.go
package repositories

import (
	"context"
	"server/domain/entities"
)

type AccountRepository interface {
	BaseRepository[*entities.Account]

	FindByEmail(ctx context.Context, email string) (*entities.Account, error)
	FindByName(ctx context.Context, name string) ([]*entities.Account, error)
}

File: modules/domain/src/repositories/account_repository.rs
use async_trait::async_trait;

// shared modules
use shared::types::DomainResponse;

// internal modules
use crate::{entities::account_entity::AccountEntity, repositories::base_repository::BaseRepository};

#[async_trait]
pub trait AccountRepository: BaseRepository<AccountEntity> {
    async fn find_by_email(&self, email: &str) -> DomainResponse<Option<AccountEntity>>;
}
